### DISCUSSION TITLE
Preconditioning Constrained Problems.

### DISCUSSION THREAD

**Discussion by TheBEllis** [idaholab/moose#29515] [Q&A Modules: Solid mechanics]
Hi moose people, I was wondering if anyone had any tips for preconditioning constrained problems? I am currently trying to use LinearNodalConstraint in an RBE3 type fashion, and I have something that works when I use full LU in serial, but as soon as I try and use boomeramg or other preconditioners, I can't get the solve to converge.

I've tried three approaches. Full LU, which works, boomeramg on the whole matrix, which doesn't converge, and a field split approach where I've preconditioned the nodes involved in the nodal constraint with LU and everything else with boomeramg. The last approach of the three, instead of just not converging, results in negative jacobians. I've attached a repo which has input files for all three approaches and my testing mesh if anyone gets time to take a look. Otherwise, any tidbits of wisdom about preconditioning constrained problems would be extremely welcome:)

https://github.com/TheBEllis/ConstraintProblem

**Comment by GiudGiud** [comment]
> The last approach of the three, instead of just not converging, results in negative jacobians

A field split approach is essentially an iteration scheme, which can diverge. 
How many equations do you have? Which variables? And which variables do the constraints affect?
